### DISCUSSION TITLE
Brainstorming for editing the paper

### DISCUSSION THREAD

**Discussion by tmke8** [wearepal/support-matching#217] [General]
## Our assumptions
- we assume that all classes, y, appear in the training set;
  in other words: we require there to be at least one source per class y in the training set
- the sources in the training set and the sources in the deployment set must together cover all sources
  that we expect to encounter at deployment time
- The number of subgroups in the deployment set needs to be known.

## Common misconceptions that we should try to avoid
- "you need to solve the assignment problem on the training set"
- "the problem is related to positive-unlabeled learning"
- "you're doing out-of-distribution generalization"
- "constructing balanced bags is computationally expensive"

### Introduction
- [ ] introduce the problem of missing sources
  - [ ] "we don't observe all outcomes for all subgroups"
  - [ ] "it is usually not all subgroups that are affected" ("privileged" group vs "disadvantaged" group)
  - [ ] we should have a canonical example here
    - [ ] something where it is clear that the subgroup membership is completely irrelevant to the task
    - [ ] ColoredMNIST fits that requirement, of course, but it's not an actually useful task
    - [ ] it should be a computer vision task
    - [ ] take the example from chapter 3 in Thomas' thesis
- [ ] introduce the concept of invariant representations
  - [ ] "it is invariant to the subgroup, so the performance on all subgroups is good"
  - [ ] "this removes information (namely the s information), but we assume that this information was irrelevant to the true task"
  - [ ] "just the training data is not adequate to learn invariant representations,
       because due to missing sources, we cannot distinguish between invariance to s and invariance to y"
  - [ ] conclude that additional information is needed to solve this problem
- [ ] argue that unlabeled data is easier to obtain than labeled data
  - [ ] "this can be seen from the popularity of unsupervised and self-supervised methods"
- [ ] explain how unlabeled data (which has all the sources!) helps with the problem
  - [ ] "the model can see what the missing source looks like. it's not just an extrapolation!"
- [ ] explain very briefly how our solution works
  - [ ] I think it's okay to be brief here because we explain it again in the theory section
  - [ ] "we use a GAN-like setup to produce a representations that makes training set and deployment set indistiguishable"
  - [ ] "in order to learn the correct invariance, we have to balance the bags in specific ways"

#### Open questions
- why not just get more data?

#### Notes
- [ ] I think we shouldn't talk about splitting z into zy and zs;
  - [ ] zs really is only needed for the reconstruction loss, so it's more of an implementation detail
       and not a crucial aspect of the method
  - [ ] it also makes figures much easier to not include this
- [ ] for the section about disentangling, we can maybe say at the beginning
     that we assume for the moment that the batches from the deployment set are balanced,
     and that we would come back to this assumption at the end of the section
- [ ] maybe mention that balancing batches is useful for other things as well
     (like normal classification problems with a strong class imbalance)
  - [ ] but maybe this is confusing to the reader
- [ ] our assumptions about the relationship of s, x and y, can be well summarized by the causal diagram s->x<-y

### Experiments
- [ ] is the performance as good as a classifier that was trained on a full dataset?

**Comment by tmke8** [comment]
Ideas for the canonical example:

- loan system with a (non-ML) preprocessing step which only really works for one demographic group, but not the other, so one demographic group is pre-filtered

**Comment by tmke8** [comment]
Relevant papers thread

**Reply by tmke8** [reply]
https://openaccess.thecvf.com/content_CVPR_2020/html/Vowels_NestedVAE_Isolating_Common_Factors_via_Weak_Supervision_CVPR_2020_paper.html

**Reply by tmke8** [reply]
https://arxiv.org/abs/1902.04601

**Comment by tmke8** [comment]
Re missing subgroup:

"“Missing subgroup” isn’t harder per se (as in more difficult to solve) it’s misspecified under our objective. Also, do we need to argue that it isn't domain adaptation?"
